src/nvim/mbyte.c

```diff
@@ -597,45 +667,41 @@ size_t mb_string2cells_len(const char *str, size_t size)
 ///
 /// @return Unicode codepoint or byte value.
 int utf_ptr2char(const char *const p_in)
-  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
+  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT FUNC_ATTR_NONNULL_ALL
 {
   uint8_t *p = (uint8_t *)p_in;
-  if (p[0] < 0x80) {  // Be quick for ASCII.
-    return p[0];
+  uint32_t v0 = p[0];
+  if (EXPECT(v0 < 0x80U, true)) {  // Be quick for ASCII.
+fret:
```

Comment:
Using `goto` with this label looks ugly. How much is this faster than using `return (int)v0` directly?

Reply:
Generated code is the same in release mode

Reply:
Hmm, then I think it's better to remove this label and use `return (int)v0` directly.